### PR TITLE
bpo-45680: Improve docs on subscriptions w.r.t. `GenericAlias` objects

### PR DESCRIPTION
**Pull request by AlexWaygood**
This PR is part of the project to improve documentation on ``GenericAlias`` objects and ``__class_getitem__``, started in #29335.

https://bugs.python.org/issue45680

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]

**Comment by AlexWaygood** [comment]
@JelleZijlstra, reckon this can be merged soon? :)

**Comment by AlexWaygood** [comment]
> Looking at this again I'm not so sure, sorry.

Thanks for taking a look! I'll have a think tomorrow about whether I want to try revising this PR, or just close it.

**Comment by AlexWaygood** [comment]
> Here, for example, there is no precise description of when `__class_getitem__` is called instead of `__getitem__`. That description exists only in the docs for `__class_getitem__`, but logically it should be part of the description of the subscript operator.

Perhaps you're right that that information might be better found in this document, but tbqh I don't have the energy to move that section from the data model to here after spending so long adding that section to the data model in GH-29389 :(

**Comment by AlexWaygood** [comment]
@JelleZijlstra I've had another bash at it. I've tried to make the section more concise and increase the accuracy of the section regarding the "expression list".

The core aims of this PR are to:

1. Make more prominent mention of the fact that subscriptions can also result in `__class_getitem__` being called, as well as `__getitem__`. 
2. Add more links to to docs elsewhere on `__class_getitem__`.

File: Doc/reference/expressions.rst [hunk 1]
```diff
@@ -815,30 +815,44 @@ Subscriptions
    object: dictionary
    pair: sequence; item
 
-Subscription of a sequence (string, tuple or list) or mapping (dictionary)
-object usually selects an item from the collection:
+The subscription of an instance of a :ref:`container class <sequence-types>`
```
Comment (JelleZijlstra) [inline Doc/reference/expressions.rst:818]: Seems weird to link this to sequence types. Maybe just drop the link? The later explanation is clearer about sequences and mappings.
```suggestion
The subscription of an instance of a container class
```

**Review by JelleZijlstra — COMMENTED** [review]
Thanks! Just one comment.

File: Doc/reference/expressions.rst (hunk 1 above)
Comment (AlexWaygood) [reply Doc/reference/expressions.rst:818]: I think the section in the docs linked to with `sequence-types` perhaps originally only discussed sequence types. Now, however, it discusses all kinds of container types -- the link takes us to here: https://docs.python.org/3/reference/datamodel.html#emulating-container-types

**Review by JelleZijlstra — APPROVED** [review]
Thanks! I'm happy to merge this now (cc @gvanrossum).

**Review by gvanrossum — APPROVED** [review]
(Because the *other* meaning of "subscription" (e.g. a magazine) is so much more common I have a slight problem with using it to refer to the a[i] syntax, but it seems commonly used in our docs so it's fine. And the unambiguous alternative, "subscripting", is used much less frequently in our docs.)

**Comment by AlexWaygood** [comment]
Thanks so much for helping get this over the line @JelleZijlstra!

**Pull request merged**